pkg/proxy/endpoints.go

```diff
@@ -109,12 +131,12 @@ func (info *BaseEndpointInfo) GetZoneHints() sets.Set[string] {
 
 // IP returns just the IP part of the endpoint, it's a part of proxy.Endpoint interface.
 func (info *BaseEndpointInfo) IP() string {
-	return proxyutil.IPPart(info.Endpoint)
+	return info.Endpoint.ip
 }
 
 // Port returns just the Port part of the endpoint.
 func (info *BaseEndpointInfo) Port() (int, error) {
-	return proxyutil.PortPart(info.Endpoint)
+	return info.Endpoint.port, nil
 }
```

Comment:
there's no need for this to still return an error, since it can't fail

Reply:
this will require modifying the `Endpoint` interface

Reply:
now it should still return an err

Reply:
> this will require modifying the `Endpoint` interface

That's fine; these are all internal APIs. You can fix all of the consumers as well.

Reply:
> > this will require modifying the `Endpoint` interface
> 
> That's fine; these are all internal APIs. You can fix all of the consumers as well.

I have resubmitted it and now I need to keep returning err.